src/layout/layout.js

```diff
@@ -284,6 +284,11 @@
       this.element_.parentElement.removeChild(this.element_);
       container.appendChild(this.element_);
 
+      var focusedElement = this.element_.querySelector('[autofocus]');
```

Comment:
Focusing and when it occurs exactly is odd between engines. I need to test the textfield itself and make sure it is functioning properly in FF.

If that checks out, then your re-ordering will be the best method to go with. Otherwise I need to patch the textfield itself and then see if that fixes the issue with layout as well.